packages/analytics-js/__tests__/components/configManager/ConfigManager.test.ts

```diff
@@ -0,0 +1,202 @@
+import { defaultHttpClient } from '@rudderstack/analytics-js/services/HttpClient';
+import { defaultErrorHandler } from '@rudderstack/analytics-js/services/ErrorHandler';
+import { defaultLogger } from '@rudderstack/analytics-js/services/Logger';
+import { ConfigManager } from '@rudderstack/analytics-js/components/configManager';
+import { state } from '@rudderstack/analytics-js/state';
+import { getSDKUrlInfo } from '@rudderstack/analytics-js/components/configManager/util/commonUtil';
+import { rest } from 'msw';
+import { CONFIG_URL, DEST_SDK_BASE_URL } from '@rudderstack/analytics-js/constants/urls';
+import { batch } from '@preact/signals-core';
+import { server } from '../../../__mocks__/msw.server';
+import { dummySourceConfigResponse } from '../../../__mocks__/fixtures';
+
+jest.mock('../../../src/services/Logger', () => {
+  const originalModule = jest.requireActual('../../../src/services/Logger');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    defaultLogger: {
+      error: jest.fn((): void => {}),
+    },
+  };
+});
+
+jest.mock('../../../src/services/ErrorHandler', () => {
+  const originalModule = jest.requireActual('../../../src/services/ErrorHandler');
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    defaultErrorHandler: {
+      onError: jest.fn((): void => {}),
+    },
+  };
+});
+
+jest.mock('../../../src/components/configManager/util/commonUtil.ts', () => {
+  const originalModule = jest.requireActual(
+    '../../../src/components/configManager/util/commonUtil.ts',
+  );
+
+  return {
+    __esModule: true,
+    ...originalModule,
+    getSDKUrlInfo: jest.fn(),
+  };
+});
+
+describe('ConfigManager', () => {
+  let configManagerInstance: ConfigManager;
+  const errorMsg = 'Unable to load the SDK due to invalid write key: " "';
+  const errorMsgSourceConfigResponse = 'Unable to fetch source config';
+  const sampleWriteKey = '2LoR1TbVG2bcISXvy7DamldfkgO';
+  const sampleDataPlaneUrl = 'https://www.dummy.url';
+  const sampleDestSDKUrl = 'https://www.sample.url/integrations';
+  const sampleConfigUrl = 'https://dummy.dataplane.host.com';
+  const sampleScriptURL = 'https://www.dummy.url/fromScript/v3/rudder-analytics.min.js';
+  const lockIntegrationsVersion = false;
+
+  const resetState = () => {
+    batch(() => {
+      state.lifecycle.writeKey.value = undefined;
+      state.lifecycle.dataPlaneUrl.value = undefined;
+      state.loadOptions.value.lockIntegrationsVersion = false;
+      state.loadOptions.value.destSDKBaseURL = DEST_SDK_BASE_URL;
+      state.loadOptions.value.logLevel = 'ERROR';
+      state.loadOptions.value.configUrl = CONFIG_URL;
+    });
+  };
+
+  beforeAll(() => {
+    server.listen();
+  });
+
+  beforeEach(() => {
+    configManagerInstance = new ConfigManager(
+      defaultHttpClient,
+      defaultErrorHandler,
+      defaultLogger,
+    );
+  });
+
+  afterEach(() => {
+    server.resetHandlers();
+    server.events.removeAllListeners();
+    resetState();
```

Comment:
What'll be the state of these fields for the first TC? Instead, can we reset the state in beforeEach method?

Reply:
the state before each test should be the default state, unless specific values need to exist for all tests.

We should clear/reset values either on after or before.

personally I would preffer to reset state on after, prepopulate on before.

Reply:
Didn't make any changes here